### PR TITLE
POST API to send alerts to Alert manager

### PR DESCRIPTION
**Pull request by rhsakarpos**
**What this PR does / why we need it**:
This API allows alerts to be pushed to Alert Manager

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
NA

**Special notes for your reviewer**:

**Release note**:

*POST API*
```http://localhost:50040/v1beta/alert```

*Request body*
```
{
    "annotations": {
      "a1": "v1",
      "a2": "v2"
    },
    "endsAt": "0001-01-01T00:00:00Z",
    "startsAt": "0001-01-01T00:00:00Z",
    "generatorURL": "http://",
    "labels": {
      "l1": "v1",
      "l2": "v2"
    }
}
```

### CODE REVIEW

**Comment by rhsakarpos** [comment]
@skdwriting @xing-yang @leonwanghui please review

File: pkg/api/controllers/alert.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+		return
+	}
+	defer resp.Body.Close()
+
+	// Marshal the result.
+	resBody, _ := json.Marshal(resp)
+	v.SuccessHandle(StatusAccepted, resBody)
+
+	return
+}
```
Comment (xing-yang) [inline pkg/api/controllers/alert.go]: Have you also implemented a GET alerts API?

File: pkg/model/alert.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+package model
+
+import "time"
+
+type LabelSet map[string]string
+
+/*
+AlertSpec is a data structure that models the alert information
+*/
```
Comment (xing-yang) [inline pkg/model/alert.go]: Suggest using "//" for consistency.

File: pkg/model/alert.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+PostableAlertSpec is a data structure that models the alert information to be sent to the Prometheus Alert Manager
+*/
+type PostableAlertSpec struct {
+
+	// annotations
+	Annotations LabelSet `json:"annotations,omitempty"`
+
+	// ends at
+	// Format: date-time
+	EndsAt time.Time `json:"endsAt,omitempty"`
```
Comment (xing-yang) [inline pkg/model/alert.go]: s/EndsAt/EndAt

File: pkg/model/alert.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+	// starts at
+	// Format: date-time
+	StartsAt time.Time `json:"startsAt,omitempty"`
```
Comment (xing-yang) [inline pkg/model/alert.go]: StartAt

File: pkg/model/alert.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+	AlertSpec
```
Comment (xing-yang) [inline pkg/model/alert.go]: Add json definition

**Comment by xing-yang** [comment]
Need a rebase.

File: pkg/api/controllers/alert.go (hunk 1 above)
Comment (rhsakarpos) [reply pkg/api/controllers/alert.go]: No...to see the alerts, we plan to re-direct from our dashboard to the Alert manager UI

File: pkg/model/alert.go (hunk 3 above)
Comment (rhsakarpos) [reply pkg/model/alert.go]: fixed

File: pkg/model/alert.go (hunk 4 above)
Comment (rhsakarpos) [reply pkg/model/alert.go]: fixed

File: pkg/model/alert.go (hunk 5 above)
Comment (rhsakarpos) [reply pkg/model/alert.go]: added

**Pull request merged**